pkg/auth/providers/azuread/azuread_client.go

```diff
@@ -0,0 +1,550 @@
+package azuread
```

Comment:
Generally, you are not closing response.Body every you should in this file. For every single http request you make, you  need to close the body.

Reply:
i noted many of the places where resp.body was not closed properly, but not all of them. make sure you double check them all